src/video/sdl_v.cpp

```diff
@@ -155,19 +164,14 @@ static void DrawSurfaceToScreen()
 	if (n == 0) return;
 
 	_num_dirty_rects = 0;
-	if (n > MAX_DIRTY_RECTS) {
-		if (_sdl_screen != _sdl_realscreen) {
-			SDL_CALL SDL_BlitSurface(_sdl_screen, NULL, _sdl_realscreen, NULL);
-		}
-		SDL_CALL SDL_UpdateRect(_sdl_realscreen, 0, 0, 0, 0);
-	} else {
-		if (_sdl_screen != _sdl_realscreen) {
-			for (int i = 0; i < n; i++) {
-				SDL_CALL SDL_BlitSurface(_sdl_screen, &_dirty_rects[i], _sdl_realscreen, &_dirty_rects[i]);
-			}
-		}
-		SDL_CALL SDL_UpdateRects(_sdl_realscreen, n, _dirty_rects);
-	}
+
+	// Just update the entire screen. Updating only the _dirty_rects
+	// seems to cause screen flickering, but that's how the SDL 1.2
+	// code worked.
+	SDL_CALL SDL_UpdateTexture(_sdl_texture, NULL, _sdl_screen->pixels, _sdl_screen->pitch);
+	SDL_CALL SDL_RenderClear(_sdl_renderer);
+	SDL_CALL SDL_RenderCopy(_sdl_renderer, _sdl_texture, NULL, NULL);
+	SDL_CALL SDL_RenderPresent(_sdl_renderer);
```

Comment:
Re this change, this means that _dirty_rects will not be used, therefore it does not need to be kept around at all?

Reply:
Yeah: the way this is currently working, `_dirty_rects` and `_num_dirty_rects` should just be replaced with a single `_dirty` or `_is_dirty` value (like [FreeRCT](https://github.com/FreeRCT/FreeRCT/blob/master/src/video.cpp)), to determine whether `DrawSurfaceToScreen()` should draw the entire screen or not. I'll make that change now.